processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -152,6 +152,6 @@ public ColumnAnalysis analyzeComplexColumn(Column column)
       size += inputSizeFn.apply(complexColumn.getRowValue(i));
     }
 
-    return new ColumnAnalysis(capabilities.getType().name(), size, null, null);
+    return new ColumnAnalysis(typeName, size, null, null);
```

Comment:
Did this need to change as well? or was this a "hey, why don't I fix this while I'm here" kind of thing?